### DISCUSSION TITLE
New files created as root:root Can this be changed in configuration

### DISCUSSION THREAD

**Discussion by norman350** [filebrowser/filebrowser#1325] [Q&A] [answered]
When I create a new directory/file they are created with root:root permissions.  Is it possible to change this to a regular user?

Thanks

**Comment by agneevX** [comment]
What's your environment?

**Reply by norman350** [reply]
I'm running the docker image from a community created template on an unraid server.  It looks like your docker run example sets the UID from the Docker run command line.  I'll see if I can tell unraid to set UID on start.

**Reply by norman350** [reply]
maybe put USER 1000 in the Dockerfile and not start the service on port 80.  port 80 requires privilege root to start.  if someone wants port 80 they can map to it.  Thanks!

**Comment by o1egl** [comment, marked as answer]
You can use `--user` docker param to set uid/gid https://filebrowser.org/installation#docker

**Reply by 0x20fearless** [reply]
> You can use `--user` docker param to set uid/gid https://filebrowser.org/installation#docker

When I set my PUID and GUID as a non root user, creating a file or folder is created with root privileges.
When creating the folder via the file system (not as root), then the folder is editable and manageable via FileBrowser without root permissions.

**Reply by agneevX** [reply]
> When I set my PUID and GUID

Those are supported by certain runtimes and aren't universal, `user` is.

**Reply by eliaskanelis** [reply]
Does not work in a Raspberry Pi. Check [this](https://github.com/filebrowser/filebrowser/discussions/1325#discussioncomment-6961110).

**Comment by Russsgithub** [comment]
.. relating to this question. 

How about if filebrowser is run with systemd service ?

I'm getting issues with after upload scripts which I think is because the file is being created as root:root.

Thanks

**Comment by eliaskanelis** [comment]
I get different results in an x86 and an Arm (Raspberry Pi) platform.

x86: Ownership **user:user**
arm: Ownership **root:root**

## Docker command

```sh
docker run -d \
            --name=${NAME} \
	    -v /mnt/HDD/server/filebrowser/data:/srv \
	    -v /mnt/HDD/server/filebrowser/config/filebrowser.db:/.filebrowser.db \
	    -v /mnt/HDD/server/filebrowser/config/filebrowser.json:/.filebrowser.json \
	    -u $(id -u):$(id -g) \
	    -p 9035:8080 \
	    --restart unless-stopped \
	    filebrowser/filebrowser
```
## Configuration

```json
{
  "port": 8080,
  "baseURL": "",
  "address": "",
  "log": "stdout",
  "database": "/.filebrowser.db",
  "root": "/srv"
}

```

**Reply by flo741** [reply]
I have the same problem in Portainer. -user:1001:1001 attribut is not working. Files can not be uploaded. DB stays zero size.
